security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -27,6 +27,11 @@ import (
 // The size of a private key for a leaf certificate.
 const keySize = 2048
 
+// SecretStore defines interface that secret storages need to implement.
+type SecretStore interface {
+	GetSecret(proxyID, token string) (*SecretItem, error)
```

Comment:
From the name indicates, the `SecretStore` is supposed to have both `Get` and `Put` methods. Reading the code, I understand it's because we send CSR in case of cache miss. And the SecretStore implementation itself have a map store, https://github.com/istio/istio/blob/collab-gcp-identity/security/pkg/nodeagent/cache/secretcache.go#L92

So it feels more like a `SecretManager`? So that the caching optimization is not visible to the users.

Reply:
thanks, renamed.